src/main/java/org/hydev/mcpm/client/arguments/ArgsParser.java

```diff
@@ -61,15 +57,20 @@ public void parse(String[] arguments) throws ArgumentParserException {
      *                                 For default handling, pass this to ArgsParser#fail.
      */
     public void parse(String[] arguments, Consumer<String> log) throws ArgumentParserException {
-        var namespace = parser.parseArgs(arguments);
+        try {
+            var namespace = parser.parseArgs(arguments);
 
-        Object handleObject = namespace.get("handler");
+            Object handleObject = namespace.get("handler");
 
-        if (!(handleObject instanceof CommandParser handler)) {
-            throw new ArgumentParserException("Unrecognized command.", parser);
-        }
+            if (!(handleObject instanceof CommandParser handler)) {
+                throw new ArgumentParserException("Unrecognized command.", parser);
+            }
 
-        handler.run(namespace, log);
+            handler.run(namespace, log);
+        }
+        catch (HelpException e) {
```

Comment:
This can't be done inside the action because... it will continue parsing arguments? Oh I see, when we set consume arguments to false, we had `--help` get consumed as the text argument. 😳